22447 Add Cooling infrastructure modeling (#22517)
netbox-community/netbox: docs/features/cooling.md

# Cooling

As part of its DCIM feature set, NetBox supports modeling data center cooling infrastructure, from facility plant down to the coolant connections on individual devices. This is used to document liquid- and hybrid-cooled environments (chillers, cooling distribution units, manifolds, rear-door heat exchangers, and cold-plate servers) as a source of truth.

## Relationship to Power

> The cooling stack deliberately mirrors the [power distribution](./power-tracking.md) model:

Reviewer: I'm not sure this is accurate any longer, given that connections among power ports & outlets are represented using cables. I would avoid referencing the power data model, and just present cooling as its own independent entity.

Reviewer 2: So not in my data centre but there is certainly scope where the cold (in) and warm (out) may not mirror.

I will add that the in and out feed in my case are not numbered the same so referencing to plumbers I couldn't say connection 22 is a cold and warm.

| Cooling | Power equivalent | Role |
|---|---|---|
| Cooling Source | Power Panel | Facility plant (the upstream origin) |
| Cooling Feed | Power Feed | A loop delivered to a rack |
| Cooling Intake | Power Port | A device-side consumer (draws coolant) |
| Cooling Outflow | Power Outlet | A device-side supply (feeds downstream equipment) |

There are two intentional differences from the power model:

- **Coolant connections are direct references, not cables.** Hoses are not structured cabling, so an intake references the outflow that supplies it directly, rather than through a cable. Tracing a loop is a walk along these references.
- **A feed represents the entire loop.** A single cooling feed covers both the supply (cold) and return (warm) paths of a loop, rather than modeling each direction separately.

## Cooling Sources

> A [cooling source](../models/dcim/coolingsource.md) is the furthest upstream cooling element modeled in NetBox, representing facility plant such as a chiller, cooling tower, dry cooler, or CRAC/CRAH unit. Each source is associated with a site, and may optionally be associated with a particular location within that site. A cooling source is not a device; it represents external facility plant, and records the coolant (fluid type) and total rated cooling capacity for the loops it originates.

Reviewer: ```suggestion
A [cooling source](../models/dcim/coolingsource.md) is the furthest upstream cooling element modeled in NetBox, representing a chiller, cooling tower, dry cooler, or CRAC/CRAH unit. Each source is associated with a site, and may optionally be associated with a particular location within that site. A cooling source is not a device; it represents external facility plant, and records the coolant (fluid type) and total rated cooling capacity for the loops it originates.
```

## Cooling Feeds

> A [cooling feed](../models/dcim/coolingfeed.md) represents a coolant loop originating from a cooling source and serving a particular rack. Each feed records an operational status, a rated cooling capacity, and a rated (design) flow rate.

Reviewer: ```suggestion
A [cooling feed](../models/dcim/coolingfeed.md) represents a coolant loop running between a cooling source and a particular rack. Each feed records an operational status, a rated cooling capacity, and a rated (design) flow rate.
```

The [cooling intakes](../models/dcim/coolingintake.md) a feed supplies are not listed on the feed explicitly; they are derived from the devices installed in the rack the feed serves. Moving a device to a different rack therefore reassigns it to that rack's feed automatically.

> 

Reviewer: Seems redundant to the section below.
```suggestion
```

## Racks and Devices

Racks and devices carry lightweight cooling attributes independent of the feed/component topology:

- A [rack](../models/dcim/rack.md) records a **cooling capability** (air-only, liquid-capable, or liquid-required) and a **cooling capacity** in kilowatts, typically inherited from its rack type.
- A [device](../models/dcim/device.md) records a **cooling method** (air, liquid, hybrid, or immersion), inherited from its device type and overridable per device.

!!! note "Scope of the initial version"
    The cooling model captures topology and design specifications. A few capabilities are intentionally left out at this point:

    - **The return path is implicit.** A single feed models the whole loop; the supply and return legs are not traced as separate objects.
    - **Path tracing and impact analysis are not yet surfaced.** The reference chain supports tracing a loop, but there is no graphical trace or "what does this source feed?" view yet.
    - **Individual hoses are not inventoried.** Connections are modeled as direct references, not as first-class cable objects.

>     - **No design pressure or temperature.** These are operating conditions with no stable design value independent of live readings, so they are recorded in a monitoring system rather than in NetBox.

Reviewer: The scope of the documentation is limited to what the application currently supports.
```suggestion
```
